src/chai/ArrayManager.cpp

```diff
@@ -137,16 +149,25 @@ void ArrayManager::move(PointerRecord* record, ExecutionSpace space)
     return;
   }
 
+  if (space == record->m_last_space) {
+    return;
+  }
+
 #if defined(CHAI_ENABLE_UM)
   if (record->m_last_space == UM) {
```

Comment:
doesn't UM have the same correctness requirements as PINNED from a synchronization perspective? Should we use the m_need_sync_for_pinned (perhaps renamed?) for UM as well?